patches/api/0318-Add-Git-information-to-version-command-on-startup.patch

```diff
@@ -0,0 +1,164 @@
++NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
++FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
++THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
++INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
++(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
++SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
++HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
++STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
++ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
++OF THE POSSIBILITY OF SUCH DAMAGE.
++*/
++package io.papermc.paper.util;
++
++import java.io.IOException;
++import java.io.InputStream;
++import java.net.URL;
++import java.util.*;
++import java.util.jar.Manifest;
++
++/**
++ * Modified version of jcabi-manifests
++ *
++ */
++public final class JarManifests {
++
++    public static final JarManifests JAR_MANIFESTS = new JarManifests();
++    public static final Map<String, String> MANIFEST_MAP;
++
++    static {
++        try {
++            MANIFEST_MAP = Collections.unmodifiableMap(JAR_MANIFESTS.getManifestMap());
++        } catch (final IOException ex) {
++            throw new RuntimeException(ex);
++        }
++    }
++
++    private JarManifests() {}
++
++    // Based on:
++    // https://github.com/jcabi/jcabi-manifests/blob/c4e1dd22bb6099769b8d279ebe3737e5b638b278/src/main/java/com/jcabi/manifests/ClasspathMfs.java#L49-L58
++
++    /**
++     * Get collection containing all META-INF/MANIFEST.MF streams
++     *
++     * @throws IOException if unable to get resources from classloader
++     */
++    public Collection<InputStream> getManifestStreams() throws IOException {
++        final Enumeration<URL> resources = getClass().getClassLoader().getResources("META-INF/MANIFEST.MF");
++        final Collection<InputStream> streams = new ArrayList<>(1);
++        while (resources.hasMoreElements()) {
++            streams.add(resources.nextElement().openStream());
++        }
++        return streams;
++    }
++
++    // Based on:
++    // https://github.com/jcabi/jcabi-manifests/blob/c4e1dd22bb6099769b8d279ebe3737e5b638b278/src/main/java/com/jcabi/manifests/Manifests.java#L209-L225
++    // https://github.com/jcabi/jcabi-manifests/blob/c4e1dd22bb6099769b8d279ebe3737e5b638b278/src/main/java/com/jcabi/manifests/Manifests.java#L381-L388
++
++    /**
++     * Get map containing entries from all manifests
++     *
++     * @throws IOException if unable to get manifest streams
++     */
++    public Map<String, String> getManifestMap() throws IOException {
++        final HashMap<String, String> attribs = new HashMap<>();
++        for (final InputStream stream : getManifestStreams()) {
++            for (final Map.Entry<Object, Object> attr: new Manifest(stream).getMainAttributes().entrySet()) {
++                attribs.put(attr.getKey().toString(), attr.getValue().toString());
```

Comment:
Should we not override the attributes in case of multiple Manifests specifying the same keys?
That's how original code did it: https://github.com/jcabi/jcabi-manifests/blob/c4e1dd22bb6099769b8d279ebe3737e5b638b278/src/main/java/com/jcabi/manifests/Manifests.java#L218-L223